projects/step-core/src/lib/shared/legacy-view-registry.service.ts

```diff
@@ -39,7 +40,14 @@ export abstract class LegacyViewRegistryService {
     template: string,
     config: { isPublicView: boolean; isStaticView: boolean }
   ): void;
-  abstract registerCustomMenuEntry(label: string, viewId: string, menuIconsClass: string, right: string): void;
+  abstract registerCustomMenuEntry(
+    label: string,
+    viewId?: string,
+    mainMenu?: boolean,
+    menuIconsClass?: string,
+    right?: string,
+    includedInMainMenu?: string
+  ): void;
```

Comment:
I don't like this functions parameters. Looks like mainMenu is boolean | undefined | null which is weird. I think you are using this to determine if it should show in the main menu. Instead I would rename "includedInMainMenu" which probably means "parentMenu" and either use "root" (maybe even in the ng2+ code as a static variable from the service) or undefined for root menue items.

Reply:
And comes to think of it, by switching to material I would rename "menuIconsClass" to "icon", but check if menuIconsClass is used somewhere else then the menu.